src/Avalonia.Controls/TabControl.cs

```diff
@@ -207,6 +207,15 @@ private void UpdateSelectedContent(Control? container = null)
                 container ??= ContainerFromIndex(SelectedIndex);
                 if (container != null)
                 {
+                    if (SelectedContentTemplate != null && ContentTemplate == null && container.GetValue(ContentTemplateProperty) != SelectedContentTemplate)
```

Comment:
The `ContentTemplate == null` check should be removed.

The code below (line 222) always uses `TabItem.ContentTemplate` if not null, so it will be applied even if the `TabControl` provides a `ContentTemplate`. Maybe add a second similar test with `TabControl.ContentTemplate` set?

Ideally the condition should be changed to exactly match the "If the value of SelectedContentTemplate is about to change" comment you've added, e.g. `if (SelectedContentTemplate != (container.GetValue(ContentTemplateProperty) ?? ContentTemplate))`